pkg/controller/endpointslice/endpointslice_tracker.go

```diff
@@ -73,33 +79,45 @@ func (est *endpointSliceTracker) Update(endpointSlice *discovery.EndpointSlice)
 	est.lock.Lock()
 	defer est.lock.Unlock()
 
-	rrv := est.relatedResourceVersions(endpointSlice)
+	rrv, _ := est.relatedResourceVersions(endpointSlice, true)
```

Comment:
Maybe instead of adding the extra bool param here, we could just add a `if !ok` and do the create here.

Reply:
Done